### PR TITLE
[IMP] pos_self_order: transparent QR code + URL revamp.

### PR DESCRIPTION
**Pull request by davidmonnom**
Before arguments in pos_self_order URL was handled with url arguments.

Now they are directly inside the URL.

TaskId: 4106628

### CODE REVIEW

File: addons/pos_self_order/controllers/self_entry.py [hunk 1]
```diff
@@ -6,8 +6,16 @@
 
 
 class PosSelfKiosk(http.Controller):
-    @http.route(["/pos-self/<config_id>", "/pos-self/<config_id>/<path:subpath>"], auth="public", website=True, sitemap=True)
-    def start_self_ordering(self, config_id=None, access_token=None, table_identifier=None):
+    @http.route(["/pos-self/<config_id>", "/pos-self/<config_id>/<path:subpath>" ], auth="public", website=True, sitemap=True)
```
Comment (vlst-odoo) [inline addons/pos_self_order/controllers/self_entry.py:9]: given that the access_token and the table identifier are optional parameters, why would we not use optional parameters?

If the goal is to make the url shorter we can just replace `access_token` with `a` and `table_identifier` with `t`, i.e. `mydb.odoo.com/pos-self?a=123&t=1`

Comment (vlst-odoo) [reply addons/pos_self_order/controllers/self_entry.py:9]: doing this means that no change to the router will be needed

Comment (davidmonnom) [reply addons/pos_self_order/controllers/self_entry.py]: need to be able to handle url

/pos-self/config_id/access_token/table_identifier/others

and

/pos_self/config_id/others?access_token=blabla&table_identifier=blabla

Comment (vlst-odoo) [reply addons/pos_self_order/controllers/self_entry.py]: you can simply have 
```py
def start_self_ordering(self, config_id=None, subpath=None, access_token=None, table_identifier=None, a=None, t=None):
    access_token = access_token or a
    table_identifier = table_identifier or t
```

Comment (davidmonnom) [reply addons/pos_self_order/controllers/self_entry.py]: see https://www.odoo.com/odoo/my-tasks/4106628

Comment (davidmonnom) [reply addons/pos_self_order/controllers/self_entry.py]: is to match the new url scheme of odoo

Comment (vlst-odoo) [reply addons/pos_self_order/controllers/self_entry.py]: then the base64 option should work well for this

**Pull request closed without merging**